xkcdpass: make available as application

das-g opened the pull request:
make `xkcdpass` available as application but keep it as a lib, too, because it [can be imported as a module](https://github.com/redacted/XKCD-password-generator/tree/xkcdpass-1.17.3#using-xkcdpass-as-an-imported-module).

###### Motivation for this change

Primary use of this Python package is as an application (CLI tool `xckdpass`).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Result of `nixpkgs-review` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>

  - xkcdpass
</details>

drewrisinger commented on pkgs/top-level/all-packages.nix:
```diff
@@ -7474,6 +7474,8 @@ in
 
   xfstests = callPackage ../tools/misc/xfstests { };
 
+  xkcdpass = with pythonPackages; toPythonApplication xkcdpass;
```
Is this the right section in ``all-packages.nix`` for this?

das-g replied:
Hmm ... maybe not.

I placed the new package by looking for [the position of `pwgen`](https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L6075-L6077) (a tool serving a similar purpose) ...
https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L6074-L6078
...  and then scrolling forward to `x...`.

However, if I search for the section titles marked by `###`, it seems that this is in section "DEVELOPMENT / EMSCRIPTEN"
https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L3132-L7826
which doesn't seem like the right category.

I guess "TOOLS" would be a better fit?
https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L506-L3131

drewrisinger reviewed (approved):
Diff LGTM, builds cleanly via ``nixpkgs-review pr 85487``

```
https://github.com/NixOS/nixpkgs/pull/85487
1 package built:
xkcdpass
```
@GrahamcOfBorg build xkcdpass

drewrisinger commented:
@GrahamcOfBorg eval
@GrahamcOfBorg build xkcdpass

drewrisinger commented:
Your original method sounded reasonable, but I think tools would probably
be a better location. I wouldn't worry about moving the other one. It
honestly doesn't matter terribly much.

On Tue, Apr 28, 2020, 6:49 PM Raphael Das Gupta <notifications@github.com>
wrote:

> *@das-g* commented on this pull request.
> ------------------------------
>
> In pkgs/top-level/all-packages.nix
> <https://github.com/NixOS/nixpkgs/pull/85487#discussion_r416969974>:
>
> > @@ -7474,6 +7474,8 @@ in
>
>    xfstests = callPackage ../tools/misc/xfstests { };
>
> +  xkcdpass = with pythonPackages; toPythonApplication xkcdpass;
>
> Hmm ... maybe not.
>
> I placed the new package by looking for the position of pwgen
> <https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L6075-L6077>
> (a tool serving a similar purpose) ...
>
> https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L6074-L6078
> ... and then scrolling forward to x....
>
> However, if I search for the section titles marked by ###, it seems that
> this is in section "DEVELOPMENT / EMSCRIPTEN"
>
> https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L3132-L7826
> which doesn't seem like the right category.
>
> I guess "TOOLS" would be a better fit?
>
> https://github.com/NixOS/nixpkgs/blob/f448ba9f6b404e2314e784a9b2d8a15c30aa3871/pkgs/top-level/all-packages.nix#L506-L3131
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/85487#discussion_r416969974>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACNZYIZRNKIFSZIHNSAVLTDRO5MPDANCNFSM4MLI3P7Q>
> .
>

das-g committed:
xkcdpass: make available as application

but keep it as a lib, too, because it
can be imported as a module:
https://github.com/redacted/XKCD-password-generator/tree/xkcdpass-1.17.3#using-xkcdpass-as-an-imported-module

das-g commented:
> Your original method sounded reasonable,

The _method_, yes, but I should have checked the _result_. :see_no_evil:

> but I think tools would probably be a better location.

I've now rebased on current `master` and moved `xkcdpass` into the "TOOLS" section:
https://github.com/NixOS/nixpkgs/blob/39269244f156254c4f08110e5d8a45a6d876286b/pkgs/top-level/all-packages.nix#L502-L3134
(Near the end of the section, due to alphabetical ordering.)

After [force-pushing that modified change](https://github.com/NixOS/nixpkgs/compare/f448ba9f6b404e2314e784a9b2d8a15c30aa3871..39269244f156254c4f08110e5d8a45a6d876286b):
Result of `nixpkgs-review pr 85487` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- xkcdpass
</details>

> I wouldn't worry about moving the other one. It honestly doesn't matter terribly much.

Yeah, left `passwd` where it currently is.

@drewrisinger Please re-review.

drewrisinger reviewed (approved):
Diff LGTM.

Builds locally via ``nixpkgs-review pr 85487``.

```
https://github.com/NixOS/nixpkgs/pull/85487
1 package built:
xkcdpass
```

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/142
